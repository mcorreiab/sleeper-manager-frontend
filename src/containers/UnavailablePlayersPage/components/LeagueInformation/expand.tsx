interface Props {
  className?: string;
}

const Hide: React.FC<Props> = ({ className }) => (
  <svg
    width="12"
    height="8"
    viewBox="0 0 12 8"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
    aria-hidden="true"
    className={className}
  >
    <path
      d="M9.875 0.99999L5.995 4.87999L2.115 0.99999C1.92817 0.812737 1.67452 0.707504 1.41 0.707504C1.14548 0.707504 0.89183 0.812737 0.704998 0.99999C0.314998 1.38999 0.314998 2.01999 0.704998 2.40999L5.295 6.99999C5.685 7.38999 6.315 7.38999 6.705 6.99999L11.295 2.40999C11.685 2.01999 11.685 1.38999 11.295 0.99999C10.905 0.61999 10.265 0.60999 9.875 0.99999Z"
      fill="#EAEAEA"
    />
  </svg>
);

export default Hide;
