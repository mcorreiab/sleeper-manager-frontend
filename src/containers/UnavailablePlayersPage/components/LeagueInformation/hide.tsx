interface Props {
  className?: string;
}

const Hide: React.FC<Props> = ({ className }) => (
  <svg
    width="12"
    height="8"
    viewBox="0 0 12 8"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
    aria-hidden="true"
    className={className}
  >
    <path
      d="M5.29501 1L0.705006 5.59C0.315006 5.98 0.315006 6.61 0.705006 7C1.09501 7.39 1.72501 7.39 2.11501 7L6.00501 3.12L9.88501 7C10.275 7.39 10.905 7.39 11.295 7C11.685 6.61 11.685 5.98 11.295 5.59L6.70501 1C6.32501 0.610004 5.68501 0.610004 5.29501 1Z"
      fill="#EAEAEA"
    />
  </svg>
);

export default Hide;
