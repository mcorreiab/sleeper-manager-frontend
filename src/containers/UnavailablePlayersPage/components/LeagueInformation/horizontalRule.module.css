.horizontalRule {
  border-width: 1px;
  border-top: 0px;
  border-color: rgba(255, 255, 255, 0.15);
  border-style: solid;
}
