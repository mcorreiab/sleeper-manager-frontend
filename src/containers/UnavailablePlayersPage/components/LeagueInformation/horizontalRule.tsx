import classnames from "classnames";
import styles from "./horizontalRule.module.css";

export interface Props {
  className?: string;
}

const horizontalRule: React.FunctionComponent<Props> = ({ className }) => (
  <hr className={classnames(styles.horizontalRule, className)} />
);

export default horizontalRule;
