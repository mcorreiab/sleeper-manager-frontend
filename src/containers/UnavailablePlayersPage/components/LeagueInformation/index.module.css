.container {
  padding: 5.2% 7.7% 0;
  background-color: #292e3c;
  border-radius: 8px;
  margin-bottom: 7.7%;
}

.container[open] {
  padding-bottom: 24px;
}

.summary {
  display: grid;
  grid-template-areas: "name expand" "situation situation";
  row-gap: 0.5rem;
}

.name {
  font-size: 1rem;
  font-weight: 700;
  grid-area: name;
}

.overview {
  font-size: 0.875rem;
  color: #8f97ab;
  grid-area: situation;
  display: flex;
  margin-bottom: 6.4%;
}

.expand {
  grid-area: expand;
  place-self: center end;
}
