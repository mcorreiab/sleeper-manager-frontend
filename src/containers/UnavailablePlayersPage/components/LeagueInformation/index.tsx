import { useState } from "react";
import ListStatusPlayer from "./listStatusPlayer";
import Hide from "./hide";
import Expand from "./expand";
import { getPlayersByStatus } from "../../rosterQuantities";
import { RosterProps } from "../../rosterProps";
import PlayersOverview from "../PlayersOverview";

export interface Props {
  roster: RosterProps;
  key?: string;
}

const LeagueInformation: React.FunctionComponent<Props> = ({ roster }) => {
  const [detailHidden, setDetailHidden] = useState(false);

  const onClick = () => {
    setDetailHidden(!detailHidden);
  };

  const arrow = detailHidden ? (
    <Hide className="col-start-2 self-center justify-self-end" />
  ) : (
    <Expand className="col-start-2 self-center justify-self-end" />
  );

  const { outPlayers, doubtfulPlayers, questionablePlayers } =
    getPlayersByStatus(roster);

  const questionableComponent =
    questionablePlayers.length > 0 ? (
      <ListStatusPlayer
        statusLabel="Questionable"
        players={questionablePlayers}
        lastStatus
      />
    ) : null;

  const doubtfulComponent =
    doubtfulPlayers.length > 0 ? (
      <ListStatusPlayer
        statusLabel="Doubtful"
        players={doubtfulPlayers}
        lastStatus={questionablePlayers.length === 0}
      />
    ) : null;

  const outComponent =
    outPlayers.length > 0 ? (
      <ListStatusPlayer
        statusLabel="Out"
        players={outPlayers}
        lastStatus={
          questionablePlayers.length === 0 && doubtfulPlayers.length === 0
        }
      />
    ) : null;

  return (
    <details
      onToggle={() => onClick()}
      className="px-5 pt-4 pb-0 bg-[#292e3c] rounded-[8px] mb-4 open:pb-5"
    >
      <summary className="grid gap-x-2 gap-y-2">
        <h2 className="text-base font-bold" key={roster.name}>
          {roster.name}
        </h2>
        {arrow}
        <PlayersOverview
          rosterName={roster.name}
          outPlayers={outPlayers.length}
          doubtfulPlayers={doubtfulPlayers.length}
          questionablePlayers={questionablePlayers.length}
          className="text-sm-lightgray mb-4"
        />
      </summary>
      {outComponent}
      {doubtfulComponent}
      {questionableComponent}
    </details>
  );
};

export default LeagueInformation;
