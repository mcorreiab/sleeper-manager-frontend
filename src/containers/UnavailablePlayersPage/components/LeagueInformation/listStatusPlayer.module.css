.container {
  display: flex;
  flex-direction: column;
}

.statusLabel {
  font-weight: 700;
  font-size: 0.75rem;
  color: #dbdfff;
  padding: 4px 8px;
  border-radius: 32px;
  background-color: #3a28a7;
  align-self: flex-start;
  margin-bottom: 0.5rem;
}

.playerList {
  list-style: none;
  font-size: 0.875rem;
  color: #8f97ab;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
}

.horizontalRule {
  margin: 0.5rem 0 1.5rem;
}
