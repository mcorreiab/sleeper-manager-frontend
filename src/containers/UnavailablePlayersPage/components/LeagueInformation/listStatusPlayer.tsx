import classnames from "classnames";
import styles from "./listStatusPlayer.module.css";
import Player from "./player";

export interface PlayerProps {
  id: string;
  name: string;
  position: string;
  team: string;
  hasHorizontalRule?: boolean;
}

export interface Props {
  statusLabel: string;
  players: PlayerProps[];
  hidden: boolean;
}

const listStatusPlayer: React.FunctionComponent<Props> = ({
  statusLabel,
  players,
  hidden,
}) => {
  const playersComponent = players.map((player, index) => (
    <Player
      key={player.id}
      name={player.name}
      position={player.position}
      team={player.team}
      hasHorizontalRule={index < players.length - 1}
    />
  ));

  return (
    <div
      className={classnames(styles.container)}
      aria-hidden={hidden}
    >
      <h3 className={styles.statusLabel}>{statusLabel}</h3>
      <ul className={styles.playerList}>{playersComponent}</ul>
    </div>
  );
};

export default listStatusPlayer;
