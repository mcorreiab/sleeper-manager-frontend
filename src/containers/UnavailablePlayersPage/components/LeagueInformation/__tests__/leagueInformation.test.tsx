import renderer from "react-test-renderer";
import LeagueInformation, { PlayerProps } from "../index";

const outPlayer = {
  id: "1",
  name: "Test Player",
  injuryStatus: "Out",
  position: "Test Position",
  team: "Test Team",
};
const questionablePlayer = {
  id: "3",
  name: "Test Player 3",
  injuryStatus: "Questionable",
  position: "Test Position 3",
  team: "Test Team 3",
};

const doubtfulPlayer = {
  id: "2",
  name: "Test Player 2",
  injuryStatus: "Doubtful",
  position: "Test Position 2",
  team: "Test Team 2",
};
it("should render league information component with all types of status", () => {
  const playersList = [outPlayer, doubtfulPlayer, questionablePlayer];

  expect(
    renderer
      .create(<LeagueInformation roster={createRoster(playersList)} />)
      .toJSON()
  ).toMatchSnapshot();
});

it("should render league information component with only out players", () => {
  expect(
    renderer
      .create(<LeagueInformation roster={createRoster([outPlayer])} />)
      .toJSON()
  ).toMatchSnapshot();
});

it("should render league information component with only questionable players", () => {
  expect(
    renderer
      .create(<LeagueInformation roster={createRoster([questionablePlayer])} />)
      .toJSON()
  ).toMatchSnapshot();
});

it("should render league information component with only doubtful players", () => {
  expect(
    renderer
      .create(<LeagueInformation roster={createRoster([doubtfulPlayer])} />)
      .toJSON()
  ).toMatchSnapshot();
});

it("should render league information component with no players", () => {
  expect(
    renderer.create(<LeagueInformation roster={createRoster([])} />).toJSON()
  ).toMatchSnapshot();
});

function createRoster(playersList: PlayerProps[]) {
  return {
    name: "Test Roster",
    size: 10,
    avatarUrl: "https://test.com",
    players: playersList,
  };
}
