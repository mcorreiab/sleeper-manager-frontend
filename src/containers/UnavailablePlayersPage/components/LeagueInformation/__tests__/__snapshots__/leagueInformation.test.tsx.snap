// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render league information component with all types of status 1`] = `
<details
  className="px-5 pt-4 pb-0 bg-[#292e3c] rounded-[8px] mb-4 open:pb-5"
  onToggle={[Function]}
>
  <summary
    className="grid gap-x-2 gap-y-2"
  >
    <h2
      className="text-base font-bold"
    >
      Test Roster
    </h2>
    <svg
      aria-hidden="true"
      className="col-start-2 self-center justify-self-end"
      fill="none"
      height="8"
      viewBox="0 0 12 8"
      width="12"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9.875 0.99999L5.995 4.87999L2.115 0.99999C1.92817 0.812737 1.67452 0.707504 1.41 0.707504C1.14548 0.707504 0.89183 0.812737 0.704998 0.99999C0.314998 1.38999 0.314998 2.01999 0.704998 2.40999L5.295 6.99999C5.685 7.38999 6.315 7.38999 6.705 6.99999L11.295 2.40999C11.685 2.01999 11.685 1.38999 11.295 0.99999C10.905 0.61999 10.265 0.60999 9.875 0.99999Z"
        fill="#EAEAEA"
      />
    </svg>
    <p
      aria-label="Test Roster players situation"
      className="text-sm row-start-2 col-span-2 flex items-center text-sm-lightgray mb-4"
    >
      1
       Out
      <svg
        aria-hidden="true"
        className="my-0 mx-1"
        fill="none"
        height="4"
        viewBox="0 0 4 4"
        width="4"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="2"
          cy="2"
          fill="#8F97AB"
          r="2"
        />
      </svg>
       
      1
       Doubtful 
      <svg
        aria-hidden="true"
        className="my-0 mx-1"
        fill="none"
        height="4"
        viewBox="0 0 4 4"
        width="4"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="2"
          cy="2"
          fill="#8F97AB"
          r="2"
        />
      </svg>
      1
       Questionable
    </p>
  </summary>
  <div
    className="flex flex-col"
  >
    <h3
      className="font-bold text-xs text-[#dbdfff] py-1 px-2 rounded-[32px] bg-sm-blue self-start mb-[0.625rem]"
    >
      Out
    </h3>
    <ul
      className="list-none text-sm text-sm-lightgray flex flex-col items-start"
    >
      <li
        className="grid grid-cols-4-autofr gap-y-2 w-full"
      >
        <p
          className="col-start-1 row-start-1 col-span-4 text-sm-lightwhite"
        >
          Test Player
        </p>
        <p
          aria-label="Test Player position"
          className="col-start-1 row-start-2"
        >
          Test Position
        </p>
        <svg
          aria-hidden="true"
          className="my-0 mx-1 col-start-2 row-start-2 self-center"
          fill="none"
          height="4"
          viewBox="0 0 4 4"
          width="4"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="2"
            cy="2"
            fill="#8F97AB"
            r="2"
          />
        </svg>
        <p
          aria-label="Test Player NFL team"
          className="col-start-3 row-start-2"
        >
          Test Team
        </p>
      </li>
    </ul>
    <hr
      className="border border-t-0 border-[#ffffff26] border-solid mt-2 mb-6 mx-0"
    />
  </div>
  <div
    className="flex flex-col"
  >
    <h3
      className="font-bold text-xs text-[#dbdfff] py-1 px-2 rounded-[32px] bg-sm-blue self-start mb-[0.625rem]"
    >
      Doubtful
    </h3>
    <ul
      className="list-none text-sm text-sm-lightgray flex flex-col items-start"
    >
      <li
        className="grid grid-cols-4-autofr gap-y-2 w-full"
      >
        <p
          className="col-start-1 row-start-1 col-span-4 text-sm-lightwhite"
        >
          Test Player 2
        </p>
        <p
          aria-label="Test Player 2 position"
          className="col-start-1 row-start-2"
        >
          Test Position 2
        </p>
        <svg
          aria-hidden="true"
          className="my-0 mx-1 col-start-2 row-start-2 self-center"
          fill="none"
          height="4"
          viewBox="0 0 4 4"
          width="4"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="2"
            cy="2"
            fill="#8F97AB"
            r="2"
          />
        </svg>
        <p
          aria-label="Test Player 2 NFL team"
          className="col-start-3 row-start-2"
        >
          Test Team 2
        </p>
      </li>
    </ul>
    <hr
      className="border border-t-0 border-[#ffffff26] border-solid mt-2 mb-6 mx-0"
    />
  </div>
  <div
    className="flex flex-col"
  >
    <h3
      className="font-bold text-xs text-[#dbdfff] py-1 px-2 rounded-[32px] bg-sm-blue self-start mb-[0.625rem]"
    >
      Questionable
    </h3>
    <ul
      className="list-none text-sm text-sm-lightgray flex flex-col items-start"
    >
      <li
        className="grid grid-cols-4-autofr gap-y-2 w-full"
      >
        <p
          className="col-start-1 row-start-1 col-span-4 text-sm-lightwhite"
        >
          Test Player 3
        </p>
        <p
          aria-label="Test Player 3 position"
          className="col-start-1 row-start-2"
        >
          Test Position 3
        </p>
        <svg
          aria-hidden="true"
          className="my-0 mx-1 col-start-2 row-start-2 self-center"
          fill="none"
          height="4"
          viewBox="0 0 4 4"
          width="4"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="2"
            cy="2"
            fill="#8F97AB"
            r="2"
          />
        </svg>
        <p
          aria-label="Test Player 3 NFL team"
          className="col-start-3 row-start-2"
        >
          Test Team 3
        </p>
      </li>
    </ul>
  </div>
</details>
`;

exports[`should render league information component with no players 1`] = `
<details
  className="px-5 pt-4 pb-0 bg-[#292e3c] rounded-[8px] mb-4 open:pb-5"
  onToggle={[Function]}
>
  <summary
    className="grid gap-x-2 gap-y-2"
  >
    <h2
      className="text-base font-bold"
    >
      Test Roster
    </h2>
    <svg
      aria-hidden="true"
      className="col-start-2 self-center justify-self-end"
      fill="none"
      height="8"
      viewBox="0 0 12 8"
      width="12"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9.875 0.99999L5.995 4.87999L2.115 0.99999C1.92817 0.812737 1.67452 0.707504 1.41 0.707504C1.14548 0.707504 0.89183 0.812737 0.704998 0.99999C0.314998 1.38999 0.314998 2.01999 0.704998 2.40999L5.295 6.99999C5.685 7.38999 6.315 7.38999 6.705 6.99999L11.295 2.40999C11.685 2.01999 11.685 1.38999 11.295 0.99999C10.905 0.61999 10.265 0.60999 9.875 0.99999Z"
        fill="#EAEAEA"
      />
    </svg>
    <p
      aria-label="Test Roster players situation"
      className="text-sm row-start-2 col-span-2 flex items-center text-sm-lightgray mb-4"
    >
      0
       Out
      <svg
        aria-hidden="true"
        className="my-0 mx-1"
        fill="none"
        height="4"
        viewBox="0 0 4 4"
        width="4"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="2"
          cy="2"
          fill="#8F97AB"
          r="2"
        />
      </svg>
       
      0
       Doubtful 
      <svg
        aria-hidden="true"
        className="my-0 mx-1"
        fill="none"
        height="4"
        viewBox="0 0 4 4"
        width="4"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="2"
          cy="2"
          fill="#8F97AB"
          r="2"
        />
      </svg>
      0
       Questionable
    </p>
  </summary>
</details>
`;

exports[`should render league information component with only doubtful players 1`] = `
<details
  className="px-5 pt-4 pb-0 bg-[#292e3c] rounded-[8px] mb-4 open:pb-5"
  onToggle={[Function]}
>
  <summary
    className="grid gap-x-2 gap-y-2"
  >
    <h2
      className="text-base font-bold"
    >
      Test Roster
    </h2>
    <svg
      aria-hidden="true"
      className="col-start-2 self-center justify-self-end"
      fill="none"
      height="8"
      viewBox="0 0 12 8"
      width="12"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9.875 0.99999L5.995 4.87999L2.115 0.99999C1.92817 0.812737 1.67452 0.707504 1.41 0.707504C1.14548 0.707504 0.89183 0.812737 0.704998 0.99999C0.314998 1.38999 0.314998 2.01999 0.704998 2.40999L5.295 6.99999C5.685 7.38999 6.315 7.38999 6.705 6.99999L11.295 2.40999C11.685 2.01999 11.685 1.38999 11.295 0.99999C10.905 0.61999 10.265 0.60999 9.875 0.99999Z"
        fill="#EAEAEA"
      />
    </svg>
    <p
      aria-label="Test Roster players situation"
      className="text-sm row-start-2 col-span-2 flex items-center text-sm-lightgray mb-4"
    >
      0
       Out
      <svg
        aria-hidden="true"
        className="my-0 mx-1"
        fill="none"
        height="4"
        viewBox="0 0 4 4"
        width="4"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="2"
          cy="2"
          fill="#8F97AB"
          r="2"
        />
      </svg>
       
      1
       Doubtful 
      <svg
        aria-hidden="true"
        className="my-0 mx-1"
        fill="none"
        height="4"
        viewBox="0 0 4 4"
        width="4"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="2"
          cy="2"
          fill="#8F97AB"
          r="2"
        />
      </svg>
      0
       Questionable
    </p>
  </summary>
  <div
    className="flex flex-col"
  >
    <h3
      className="font-bold text-xs text-[#dbdfff] py-1 px-2 rounded-[32px] bg-sm-blue self-start mb-[0.625rem]"
    >
      Doubtful
    </h3>
    <ul
      className="list-none text-sm text-sm-lightgray flex flex-col items-start"
    >
      <li
        className="grid grid-cols-4-autofr gap-y-2 w-full"
      >
        <p
          className="col-start-1 row-start-1 col-span-4 text-sm-lightwhite"
        >
          Test Player 2
        </p>
        <p
          aria-label="Test Player 2 position"
          className="col-start-1 row-start-2"
        >
          Test Position 2
        </p>
        <svg
          aria-hidden="true"
          className="my-0 mx-1 col-start-2 row-start-2 self-center"
          fill="none"
          height="4"
          viewBox="0 0 4 4"
          width="4"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="2"
            cy="2"
            fill="#8F97AB"
            r="2"
          />
        </svg>
        <p
          aria-label="Test Player 2 NFL team"
          className="col-start-3 row-start-2"
        >
          Test Team 2
        </p>
      </li>
    </ul>
  </div>
</details>
`;

exports[`should render league information component with only out players 1`] = `
<details
  className="px-5 pt-4 pb-0 bg-[#292e3c] rounded-[8px] mb-4 open:pb-5"
  onToggle={[Function]}
>
  <summary
    className="grid gap-x-2 gap-y-2"
  >
    <h2
      className="text-base font-bold"
    >
      Test Roster
    </h2>
    <svg
      aria-hidden="true"
      className="col-start-2 self-center justify-self-end"
      fill="none"
      height="8"
      viewBox="0 0 12 8"
      width="12"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9.875 0.99999L5.995 4.87999L2.115 0.99999C1.92817 0.812737 1.67452 0.707504 1.41 0.707504C1.14548 0.707504 0.89183 0.812737 0.704998 0.99999C0.314998 1.38999 0.314998 2.01999 0.704998 2.40999L5.295 6.99999C5.685 7.38999 6.315 7.38999 6.705 6.99999L11.295 2.40999C11.685 2.01999 11.685 1.38999 11.295 0.99999C10.905 0.61999 10.265 0.60999 9.875 0.99999Z"
        fill="#EAEAEA"
      />
    </svg>
    <p
      aria-label="Test Roster players situation"
      className="text-sm row-start-2 col-span-2 flex items-center text-sm-lightgray mb-4"
    >
      1
       Out
      <svg
        aria-hidden="true"
        className="my-0 mx-1"
        fill="none"
        height="4"
        viewBox="0 0 4 4"
        width="4"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="2"
          cy="2"
          fill="#8F97AB"
          r="2"
        />
      </svg>
       
      0
       Doubtful 
      <svg
        aria-hidden="true"
        className="my-0 mx-1"
        fill="none"
        height="4"
        viewBox="0 0 4 4"
        width="4"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="2"
          cy="2"
          fill="#8F97AB"
          r="2"
        />
      </svg>
      0
       Questionable
    </p>
  </summary>
  <div
    className="flex flex-col"
  >
    <h3
      className="font-bold text-xs text-[#dbdfff] py-1 px-2 rounded-[32px] bg-sm-blue self-start mb-[0.625rem]"
    >
      Out
    </h3>
    <ul
      className="list-none text-sm text-sm-lightgray flex flex-col items-start"
    >
      <li
        className="grid grid-cols-4-autofr gap-y-2 w-full"
      >
        <p
          className="col-start-1 row-start-1 col-span-4 text-sm-lightwhite"
        >
          Test Player
        </p>
        <p
          aria-label="Test Player position"
          className="col-start-1 row-start-2"
        >
          Test Position
        </p>
        <svg
          aria-hidden="true"
          className="my-0 mx-1 col-start-2 row-start-2 self-center"
          fill="none"
          height="4"
          viewBox="0 0 4 4"
          width="4"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="2"
            cy="2"
            fill="#8F97AB"
            r="2"
          />
        </svg>
        <p
          aria-label="Test Player NFL team"
          className="col-start-3 row-start-2"
        >
          Test Team
        </p>
      </li>
    </ul>
  </div>
</details>
`;

exports[`should render league information component with only questionable players 1`] = `
<details
  className="px-5 pt-4 pb-0 bg-[#292e3c] rounded-[8px] mb-4 open:pb-5"
  onToggle={[Function]}
>
  <summary
    className="grid gap-x-2 gap-y-2"
  >
    <h2
      className="text-base font-bold"
    >
      Test Roster
    </h2>
    <svg
      aria-hidden="true"
      className="col-start-2 self-center justify-self-end"
      fill="none"
      height="8"
      viewBox="0 0 12 8"
      width="12"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9.875 0.99999L5.995 4.87999L2.115 0.99999C1.92817 0.812737 1.67452 0.707504 1.41 0.707504C1.14548 0.707504 0.89183 0.812737 0.704998 0.99999C0.314998 1.38999 0.314998 2.01999 0.704998 2.40999L5.295 6.99999C5.685 7.38999 6.315 7.38999 6.705 6.99999L11.295 2.40999C11.685 2.01999 11.685 1.38999 11.295 0.99999C10.905 0.61999 10.265 0.60999 9.875 0.99999Z"
        fill="#EAEAEA"
      />
    </svg>
    <p
      aria-label="Test Roster players situation"
      className="text-sm row-start-2 col-span-2 flex items-center text-sm-lightgray mb-4"
    >
      0
       Out
      <svg
        aria-hidden="true"
        className="my-0 mx-1"
        fill="none"
        height="4"
        viewBox="0 0 4 4"
        width="4"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="2"
          cy="2"
          fill="#8F97AB"
          r="2"
        />
      </svg>
       
      0
       Doubtful 
      <svg
        aria-hidden="true"
        className="my-0 mx-1"
        fill="none"
        height="4"
        viewBox="0 0 4 4"
        width="4"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="2"
          cy="2"
          fill="#8F97AB"
          r="2"
        />
      </svg>
      1
       Questionable
    </p>
  </summary>
  <div
    className="flex flex-col"
  >
    <h3
      className="font-bold text-xs text-[#dbdfff] py-1 px-2 rounded-[32px] bg-sm-blue self-start mb-[0.625rem]"
    >
      Questionable
    </h3>
    <ul
      className="list-none text-sm text-sm-lightgray flex flex-col items-start"
    >
      <li
        className="grid grid-cols-4-autofr gap-y-2 w-full"
      >
        <p
          className="col-start-1 row-start-1 col-span-4 text-sm-lightwhite"
        >
          Test Player 3
        </p>
        <p
          aria-label="Test Player 3 position"
          className="col-start-1 row-start-2"
        >
          Test Position 3
        </p>
        <svg
          aria-hidden="true"
          className="my-0 mx-1 col-start-2 row-start-2 self-center"
          fill="none"
          height="4"
          viewBox="0 0 4 4"
          width="4"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="2"
            cy="2"
            fill="#8F97AB"
            r="2"
          />
        </svg>
        <p
          aria-label="Test Player 3 NFL team"
          className="col-start-3 row-start-2"
        >
          Test Team 3
        </p>
      </li>
    </ul>
  </div>
</details>
`;
